docs/01_nodeos/02_usage/03_development-environment/10_local-single-node-testnet-consensus.md

```diff
@@ -0,0 +1,181 @@
+5. [Install eosio.boot System Contract](#5-install-eosio.boot-system-contract)
+6. [Activate The Remaining Protocol Features](#6-activate-the-remaining-protocol-features)
+7. [Install eosio.bios System Contract](#7-install-eosio.bios-system-contract)
+
+### 1. Add the development key to the wallet
+
+Execute the following at the command prompt:
+
+```sh
+cleos wallet import --private-key 5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3
+```
+
+### 2. Start the Producer Node
+
+Start your own single-node blockchain with this single command:
+
+```sh
+nodeos -e -p eosio --plugin eosio::chain_api_plugin --plugin eosio::history_api_plugin
+```
+
+[[info | Nodeos Minimal Options]]
+| A minimal `nodeos` instance setup for block production requires both `chain_api_plugin` and `history_api_plugin` with the `-e` option (enable stale production) and `-p eosio` option (producer name `eosio`). Alternatively, you can also setup and specify your own account as the producer name.
+
+After running `nodeos`, you should get log messages similar as below. It means the blocks are successfully produced.
+
+```console
+1575001ms thread-0   chain_controller.cpp:235      _push_block          ] initm #1 @2017-09-04T04:26:15  | 0 trx, 0 pending, exectime_ms=0
+1575001ms thread-0   producer_plugin.cpp:207       block_production_loo ] initm generated block #1 @ 2017-09-04T04:26:15 with 0 trxs  0 pending
+1578001ms thread-0   chain_controller.cpp:235      _push_block          ] initc #2 @2017-09-04T04:26:18  | 0 trx, 0 pending, exectime_ms=0
+1578001ms thread-0   producer_plugin.cpp:207       block_production_loo ] initc generated block #2 @ 2017-09-04T04:26:18 with 0 trxs  0 pending
+...
+eosio generated block 046b9984... #101527 @ 2018-04-01T14:24:58.000 with 0 trxs
+eosio generated block 5e527ee2... #101528 @ 2018-04-01T14:24:58.500 with 0 trxs
+...
+```
+
+At this point, `nodeos` is running with a single producer, `eosio`.
+
+### 3. Preactivate Protocol Features
+
+All of the protocol upgrade features introduced in v1.8 and on subsequent versions require a special protocol feature (codenamed `PREACTIVATE_FEATURE`) to be activated and for an updated version of the system contract that makes use of the functionality introduced by that feature to be deployed.
+
+To activate the special protocol `PREACTIVATE_FEATURE` run the following command:
+
+```sh
+curl --request POST \
+    --url http://127.0.0.1:8888/v1/producer/schedule_protocol_feature_activations \
+    -d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}'
+```
+
+### 4. Build System Smart Contracts
+
+Two reference system smart contracts, the `eosio.boot` and `eosio.bios`, are located in the [`eos`](https://github.com/EOSIO/eos.git) repository. To build them execute the following at the command prompt:
+
+```sh
+cd ~
+git clone https://github.com/EOSIO/eos.git
+cd ./eos/contracts/contracts/
+mkdir build
+cd build
+cmake ..
+make
```

Comment:
They are included pre-made in the repo. So they could just use them without building. To build they are going to need CDT installed and setup.

Reply:
ah I did not notice we checked in the binaries.